src/utils/scopeexit.hpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SCOPE_EXIT_HPP_
+#define SCOPE_EXIT_HPP_
+
+#include <functional>
+
+/**
+ * Scope Exit.
```

Comment:
Please add more informative description like: An object to call given function on object destruction.

Reply:
Removed